app/models/service_ovf_template.rb

```diff
@@ -0,0 +1,149 @@
+class ServiceOvfTemplate < ServiceGeneric
+  delegate :template, :manager, :to => :service_template, :allow_nil => true
+
+  # A chance for taking options from automate script to override options from a service dialog
+  def preprocess(action, new_options = {})
+    return unless action == ResourceAction::PROVISION
+
+    if new_options.present?
+      _log.info("Override with new options:")
+      $log.log_hashes(new_options)
+    end
+
+    save_action_options(action, new_options)
+  end
+
+  def execute(action)
+    return unless action == ResourceAction::PROVISION
+
+    library_item_deploy_queue
+  end
+
+  def library_item_deploy_queue
+    task_options = {:action => "Deploying VMware Content Library Item", :userid => "system"}
```

Comment:
Shouldn't this live in the VMware repo?

Reply:
That is the options required by `MiqTask.generic_action_with_callback`.

Reply:
This whole class should live in the vmware repo since it is filled with VMware specifics, e.g. [:point_down:](https://github.com/ManageIQ/manageiq/pull/20555/files#diff-4b3bb85beddbe4cdbaad83bee45b5c3eR45)